test/Common/DurableTaskEndToEndTests.cs

```diff
@@ -2923,6 +2923,110 @@ public async Task DurableEntity_EntityNameCaseInsensitivity(bool extendedSession
             }
         }
 
+        [Fact]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        public async Task AzureStorage_TimerLimitExceeded_ThrowsException()
+        {
+            string orchestrationFunctionName = nameof(TestOrchestrations.SimpleTimerSucceeds);
+
+            using (var host = TestHelpers.GetJobHost(
+                this.loggerProvider,
+                nameof(this.AzureStorage_TimerLimitExceeded_ThrowsException),
+                false))
+            {
+                await host.StartAsync();
+
+                var invalidFireAtTime = DateTime.UtcNow.AddDays(7);
+
+                var client = await host.StartOrchestratorAsync(orchestrationFunctionName, invalidFireAtTime, this.output);
+
+                var status = await client.WaitForCompletionAsync(this.output);
+
+                Assert.Equal(OrchestrationRuntimeStatus.Completed, status.RuntimeStatus);
+
+                Assert.False((bool)status.Output);

```

Comment:
Consider having the orchestration fail with an exception, and then checking the output for some expected part of the exception message, like invalid parameter name. That would ensure we don't have false positives caused by unrelated `ArgumentException`s.